Filtered reports generation

elasticspoon opened the pull request:
Issue https://github.com/rubyforgood/casa/issues/5579
Blocked by https://github.com/rubyforgood/casa/pull/5590
Fixes https://github.com/rubyforgood/casa/issues/3243 (i think? i just randomly found this issue)

To close issue still need to implement adding court topics, etc to reports themselves. 

Reworks report generation page to use a modal and allow selection of contacts into report by date range.

New generate page with new button:
![image](https://github.com/rubyforgood/casa/assets/14540596/e6a98566-1257-42ea-b446-e12f39db8c29)

New modal with generation options:
![image](https://github.com/rubyforgood/casa/assets/14540596/ae374267-0ff9-4369-aa36-d1f54ff07408)
![image](https://github.com/rubyforgood/casa/assets/14540596/cb613c95-ab48-41a5-bef5-c79d30e8a219)

Modal on both generation pages:
![image](https://github.com/rubyforgood/casa/assets/14540596/a75cf07b-760f-4436-be22-ad97b4cbeff0)

elasticspoon committed:
feat: rework court report generation

Add generate docx modal with date pickers to filter dates of case
contacts to include in the generated report.

Modify current tests to open the modal before
attempting to click within it.

Allow range filtering of contacts in case report

elasticspoon committed:
refactor: expose court report methods

Expose methods in court report context to make unit
testing easier.

Write additional unit tests.
Remove old tests.

Remove redundant tests, some behavior is
already tested by the Dates object.

compwron reviewed (approved):
I am in favor - I want to look at this more deeply when I have time. @FireLemons do you have time to review?

elasticspoon committed:
fix: improved query

elasticspoon committed:
feat: use most recent court date in modal

On the case show page it should autofill the start date of
the date range with the most recent court date.

This most closely mirrors existing behavior.

compwron commented on app/models/casa_case.rb:
```diff
@@ -158,6 +158,12 @@ def most_recent_past_court_date
     court_dates.where("date < ?", Date.today).order(:date).last
   end
 
+  def formatted_latest_court_date
```
has test?

compwron commented on spec/system/casa_cases/show_spec.rb:
```diff
@@ -41,6 +41,58 @@
     end
   end
 
+  describe "Report Generation", js: true do
```
YAY test

compwron commented on spec/system/case_court_reports/index_spec.rb:
```diff
@@ -204,6 +171,7 @@
     end
   end
 
+  # TODO: make this a request spec
```
agreed

compwron commented:
changing reports generation makes me very nervous but I think this is reasonable

compwron commented:
screenshots are good... let's do it
